lib/driver.js

```diff
@@ -557,7 +557,10 @@ class XCUITestDriver extends BaseDriver {
       }
     }
 
-
+    // there is no point in having `keychainPath` without `keychainPassword`
+    if ((caps.keychainPath && !caps.keychainPassword) || (!caps.keychainPath && caps.keychainPassword)) {
```

Comment:
oops sorry, got the polarity wrong, should just be:

```js
if (!!caps.keychainPath ^ !!caps.keychainPassword) {
```

since `^` returns true iff the boolean operands are different, which is the case we want here.